riocli/utils/execute.py

```diff
@@ -62,12 +31,24 @@ def run_on_device(
         user: str = 'root',
         shell: str = '/bin/bash',
         background: bool = False,
+        deployment: str = None,
+        exec_name: str = None
 ) -> str:
     client = new_client()
     device = client.get_device(device_id=device_guid)
+
+    client = new_v2_client()
+    package = client.get_package(deployment.metadata.depends.nameOrGUID, query={"version": deployment.metadata.depends.version})
+
+    if exec_name is None:
+        executables = [e.name for e in package.spec.executables]
+        exec_name = show_selection(executables, '\nChoose the executable')
```

Comment:
todo: Move the exec selection to the `rio deployment execute` command. Make this method throw an error when `exec_name` is `None` when the command is supposed to run on a deployment.